doc/espnet2_tutorial.md

```diff
@@ -315,6 +315,60 @@ To use SSLRs in your task, you need to make several modifications.
    ```
 3. Because the shift sizes of different `upstream` models are different, e.g. `HuBERT` and `Wav2Vec2.0` have `20ms` frameshift. Sometimes, the downsampling rate (`input_layer`) in the `encoder` configuration need to be changed. For example, using `input_layer: conv2d2` will results in a total frameshift of `40ms`, which is enough for some tasks.
 
+## Usage of adapters in ESPnet2
+
+Following the previous introduction on SSLR in ESPnet, we will introduce several options to use adapters in ESPnet and leverage the representation brought by SSLR. Available options are:
+- Insert adapters to specific layers with a pre-trained s3prl frontend for adaptation of downstream tasks*.
+- Enable automatic adapter insertion via `FindAdaptNet`.
+
+Note this is done for model training, so modifications would be at __stage 11__ of ESPnet2 recipes.
+### Prerequisites
+1. Install [S3PRL](https://github.com/s3prl/s3prl) by `tools/installers/install_s3prl.sh`.
+2. Wav2Vec is needed, [fairseq](https://github.com/pytorch/fairseq) should be installed by `tools/installers/install_fairseq.sh`.
```

Comment:
Can you double check this? I think the s3prl now doesn't depend on the fairseq to use adapters. If necessary, you may need to update your adapter code a bit.

Reply:
Thanks for the catch! Yes, indeed the fairseq part is unnecessary since s3prl has a prototype of `TransformerSentenceEncoderLayer` in their codebase which I can refer to when testing the whole module (Originally importing fairseq was for using the `TransformerSentenceEncoderLayer` at testing stage). Will remove this prereq in the next commit